binstar_client/commands/login.py

```diff
@@ -4,17 +4,57 @@
 
 import getpass
 import logging
+import os
 import platform
+import re
 import socket
 import sys
-from urllib.parse import urlparse
+from urllib.parse import urlparse, urljoin
 
+from anaconda_auth.actions import _do_auth_flow
+from anaconda_auth.config import AnacondaAuthSite
+from anaconda_auth.token import TokenInfo
+from anaconda_auth.exceptions import TokenNotFoundError
 from binstar_client import errors
 from binstar_client.utils import get_config, get_server_api, store_token, bool_input
 
+LEGACY_INTERACTIVE_LOGIN = os.getenv("ANACONDA_CLIENT_LEGACY_INTERACTIVE_LOGIN", False)
+
 logger = logging.getLogger('binstar.login')
 
 
+class DotOrgSite(AnacondaAuthSite):
+    @property
+    def domain_for_keyring_token(self):
+        # The unified CLI/SDK does not prefer the "auth." subdomain
+        # API Keys generated from `anaconda login` are stored in the
+        # keyring without the "auth." subdomain.
+        domain = re.sub(r"^auth\.", "", urlparse(self.oidc.authorization_endpoint).netloc)
+        return domain
+
+    @property
+    def login_success_url(self) -> str:
+        """The location to redirect after auth flow, if successful."""
+        return urljoin(f"https://{self.domain_for_keyring_token}", self.login_success_path)
```

Comment:
@CodingBurak this is where the URL comes from for the success page. Let's ensure we have in place a solution to make sure this exists, or at least a redirect does.